Assets/MixedRealityToolkit.Tests/PlayModeTests/SpeechTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// The assembly that this file is part of is still built for the player,
+// even though the assembly itself is marked as a test assembly (this is not
+// expected because test assemblies should not be included in player builds).
+// Because the .NET backend is deprecated in 2018 and removed in 2019 and this
+// issue will likely persist for 2018, this issue is worked around by wrapping all
+// play mode tests in this check.
+
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Linq;
+using Microsoft.MixedReality.Toolkit.Diagnostics;
+using System.Runtime.InteropServices;
+using System;
+using Microsoft.MixedReality.Toolkit.UI;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    public class SpeechTests
+    {
+        [SetUp]
+        public void Setup()
+        {
+            PlayModeTestUtilities.Setup();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            PlayModeTestUtilities.TearDown();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestToggleProfilerCommand()
+        {
+            // Confirm that the Diagnostics system is enabled and the VisualProfiler is in the scene.
+            IMixedRealityDiagnosticsSystem diagnosticsSystem = null;
+            MixedRealityServiceRegistry.TryGetService<IMixedRealityDiagnosticsSystem>(out diagnosticsSystem);
+            Assert.IsNotNull(diagnosticsSystem, "The diagnostics system is not enabled in the scene.");
+            yield return null;
+
+            // Verfiy that the VisualProfiler is enabled.
+            Assert.IsTrue(diagnosticsSystem.ShowProfiler, "The VisualProfiler is not active.");
+            yield return null;
+
+            // Toggle the profiler visualization off.
+            KeyboardEvent(VirtualKey_9, ScanCode_9, 0, IntPtr.Zero); // key down
+            yield return null;
+            KeyboardEvent(VirtualKey_9, ScanCode_9, KeyboardFlag_KeyUp, IntPtr.Zero); // key up
+            for (int i = 0; i < 10; i++) { yield return null; }
```

Comment:
What's the significance of these 10 yields?

Reply:
timing to allow the system state to be updated by the speech event handler.